controllers/helpers.go

```diff
@@ -39,9 +39,20 @@ const (
 func CreateNutanixClient(ctx context.Context, client client.Client, nutanixCluster *infrav1.NutanixCluster) (*nutanixClientV3.Client, error) {
 	creds, err := nutanixClientHelper.GetConnectionInfo(client, ctx, nutanixCluster)
 	if err != nil {
+		klog.Errorf("Error getting connection info for creating Nutanix client: %v", err)
 		return nil, err
 	}
-	return nutanixClientHelper.Client(*creds, nutanixClientHelper.ClientOptions{})
+	c, err := nutanixClientHelper.Client(*creds, nutanixClientHelper.ClientOptions{})
+	if err != nil {
+		klog.Errorf("Error creating Nutanix client: %v", err)
```

Comment:
If you're returning the error; leave the choice of logging it to the caller. Here's a well-thought-out blog post about logging. https://dave.cheney.net/2015/11/05/lets-talk-about-logging